backend/fbdev.c

```diff
@@ -56,12 +57,23 @@ static void _twin_fbdev_put_span(twin_coord_t left,
         return;
 
     twin_coord_t width = right - left;
-    off_t off = top * screen->width + left;
-    uint32_t *dest =
-        (uint32_t *) ((uintptr_t) tx->fb_base + (off * sizeof(uint32_t)));
+    off_t off = tx->fb_fix.line_length * top + 4 * left;
+    unsigned char *dest = (unsigned char *) tx->fb_base + off;
     memcpy(dest, pixels, width * sizeof(uint32_t));
 }
 
+static void twin_fbdev_cache_sync(void *closure)
+{
+    twin_fbdev_t *tx = PRIV(closure);
+    unsigned int args[2];
+    args[0] = tx->fb_base;
+    args[1] = tx->fb_base + tx->fb_var.yres_virtual * tx->fb_fix.line_length;
+    ioctl(tx->fb_fd, FBIO_CACHE_SYNC, args);
```

Comment:
You should explain why `FBIO_CACHE_SYNC` is required.

Reply:
This is driver-specificed code and should not be used in here, I will remove this code.